tests/phpunit/CRM/Civicase/Hook/PostProcess/AttachEmailActivityToAllCasesTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+use CRM_Civicase_Hook_PostProcess_AttachEmailActivityToAllCases as AttachEmailActivityToAllCases;
+use CRM_Civicase_Test_Fabricator_Case as CaseFabricator;
+use CRM_Civicase_Test_Fabricator_CaseType as CaseTypeFabricator;
+use CRM_Civicase_Test_Fabricator_Contact as ContactFabricator;
+
+/**
+ * Tests for the CRM_Civicase_Hook_PostProcess_AttachEmailActivityToAllCases.
+ *
+ * @group headless
+ */
+class CRM_Civicase_Hook_PostProcess_AttachEmailActivityToAllCasesTest extends BaseHeadlessTest {
+
+  /**
+   * Test the run method.
+   *
+   * @param int $activityId
+   *   Activity id.
+   * @param int $caseId
+   *   First case id.
+   * @param string $allCaseIds
+   *   All case ids.
+   *
+   * @dataProvider getTestDataForRunMethod
+   */
+  public function testRun($activityId, $caseId, $allCaseIds) {
+    $_GET['caseid'] = $_REQUEST['caseid'] = $caseId;
+    $_GET['allCaseIds'] = $_REQUEST['allCaseIds'] = $allCaseIds;
+    $hook = new AttachEmailActivityToAllCases();
+    $hook->run(CRM_Contact_Form_Task_Email::class, new CRM_Contact_Form_Task_Email());
+    $allCaseIds = array_diff(explode(',', $allCaseIds), [$caseId]);
+    foreach ($allCaseIds as $caseId) {
+      $activity = civicrm_api3('Activity', 'get', [
+        'case_id' => $caseId,
+        'id' => $activityId,
+        'options' => [
+          'limit' => 1,
+        ],
```

Comment:
No need to add this part
```php
 'options' => [
          'limit' => 1,
        ],
```
Part of the tests should be that there is no duplicate activity created for a case. So we can assert that only one activity record is returned.

Reply:
removed